src/Data/Parity/HomomorphismNatParity.agda

```diff
@@ -0,0 +1,103 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some properties about parities
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Data.Parity.HomomorphismNatParity where
+
+open import Data.Nat.Base
+  using (zero; suc; _+_; _*_; parity
+  ; +-rawMagma; +-0-rawMonoid
+  ; +-*-rawNearSemiring; +-*-rawSemiring)
+open import Data.Parity.Base using (Parity; 0ℙ; 1ℙ; _⁻¹)
+  renaming (_+_ to _+ℙ_; _*_ to _*ℙ_
+  ; +-rawMagma to ℙ+-rawMagma; +-0-rawMonoid to ℙ+-0-rawMonoid
+  ; +-*-rawNearSemiring to ℙ+-*-rawNearSemiring; +-*-rawSemiring to ℙ+-*-rawSemiring)
+open import Data.Parity.Properties using (⁻¹-inverts)
+open import Relation.Binary.PropositionalEquality
+  using (_≡_; refl; sym; cong; cong₂; module ≡-Reasoning)
+
+open import Algebra.Morphism.Structures
+  using (IsMagmaHomomorphism; IsMonoidHomomorphism
+  ; IsNearSemiringHomomorphism; IsSemiringHomomorphism)
+
+------------------------------------------------------------------------
+-- relating Nat and Parity
+
+-- successor and (_⁻¹)
+
+suc-homo-⁻¹ : ∀ n → (parity (suc n)) ⁻¹ ≡ parity n
+suc-homo-⁻¹ zero    = refl
+suc-homo-⁻¹ (suc n) = ⁻¹-inverts (suc-homo-⁻¹ n)
+
+-- parity is a _+_ homomorphism
+
++-homo-+      : ∀ m n → parity (m + n) ≡ (parity m) +ℙ (parity n)
+suc-+-homo-⁻¹ : ∀ m n → parity (suc m + n) ≡ (parity m) ⁻¹ +ℙ (parity n)
+
+suc-+-homo-⁻¹ zero    n = sym (suc-homo-⁻¹ (suc n))
+suc-+-homo-⁻¹ (suc m) n = begin
+  parity (suc (suc m) + n)          ≡⟨⟩
+  parity (m + n)                    ≡⟨ +-homo-+ m n ⟩
+  (parity m) +ℙ (parity n)          ≡⟨ cong (_+ℙ (parity n)) (sym (suc-homo-⁻¹ m)) ⟩
+  (parity (suc m)) ⁻¹ +ℙ (parity n) ∎  where open ≡-Reasoning
+
++-homo-+ zero    n = refl
++-homo-+ (suc m) n = begin
+  parity (suc m + n)          ≡⟨ suc-+-homo-⁻¹ m n ⟩
+  (parity m) ⁻¹ +ℙ (parity n) ≡⟨ cong (_+ℙ (parity n)) (suc-homo-⁻¹ (suc m)) ⟩
+  parity (suc m) +ℙ parity n  ∎ where open ≡-Reasoning
+
+-- parity is a _*_ homomorphism
+
+*-homo-*      : ∀ m n → parity (m * n) ≡ (parity m) *ℙ (parity n)
+*-homo-* zero    n = refl
+*-homo-* (suc m) n with parity m in m≡p | parity n in n≡q
```

Comment:
We're not pattern matching on the result of `with` so why not simply define `p = parity m` in the `where` block? Then no need for `m≡p` etc.

Reply:
You're absolutely right. Fixed now.